docs/usage/upload_files.rst

```diff
@@ -0,0 +1,199 @@
+..
+    This file is part of Invenio.
+    Copyright (C) 2015-2019 CERN.
+
+    Invenio is free software; you can redistribute it and/or modify it
+    under the terms of the MIT License; see LICENSE file for more details.
+
+
+Creating Buckets
+----------------
+
+In order to upload, modify or delete files, we first need to create a bucket.
+A bucket can be created by a POST request to the url prefix, i.e. /files.
+The response will contain the unique ID of the bucket.
+A bucket can have one or more tags which store extra metadata for that bucket.
+Each tag is uniquely identified by a key.
+
+First we create a bucket:
+
+.. code-block:: console
+
+   $ curl -X POST http://localhost:5000/files
+
+.. code-block:: json
+
+    {
+        "max_file_size": null,
+        "updated": "2019-05-16T13:07:21.595398+00:00",
+        "locked": false,
+        "links": {
+            "self": "http://localhost:5000/files/cb8d0fa7-2349-484b-89cb-16573d57f09e",
+            "uploads": "http://localhost:5000/files/cb8d0fa7-2349-484b-89cb-16573d57f09e?uploads",
+            "versions": "http://localhost:5000/files/cb8d0fa7-2349-484b-89cb-16573d57f09e?versions"
+        },
+        "created": "2019-05-16T13:07:21.595391+00:00",
+        "quota_size": null,
+        "id": "cb8d0fa7-2349-484b-89cb-16573d57f09e",
+        "size": 0
+    }
+
+Upload Files
+------------
+
+The REST API allows you to upload, download and modify single files.
+A file is uniquely identified within a bucket by its name.
+Each file can have multiple versions.
+
+Let's upload a file called my_file.txt inside the bucket we just created.
+To do this we need to use the PUT command, and specify the bucket ID as well as the file ("my_file.txt")
+
+Upload a file:
+
+.. code-block:: console
+
+   $ B=cb8d0fa7-2349-484b-89cb-16573d57f09e
+
+   $ curl -i -X PUT --data-binary @my_file.txt \
+     "http://localhost:5000/files/$B/my_file.txt"
+
+.. code-block:: json
+
+    {
+        "mimetype": "text/plain",
+        "updated": "2019-05-16T13:10:22.621533+00:00",
+        "links": {
+            "self": "http://localhost:5000/files/cb8d0fa7-2349-484b-89cb-16573d57f09e/my_file.txt",
+            "version": "http://localhost:5000/files/cb8d0fa7-2349-484b-89cb-16573d57f09e/my_file.txt?versionId=7f62676d-0b8e-4d77-9687-8465dc506ca8",
+            "uploads": "http://localhost:5000/files/cb8d0fa7-2349-484b-89cb-16573d57f09e/my_file.txt?uploads"
+        },
+        "is_head": true,
+        "tags": {},
+        "checksum": "md5:d7d02c7125bdcdd857eb70cb5f19aecc",
+        "created": "2019-05-16T13:10:22.617714+00:00",
+        "version_id": "7f62676d-0b8e-4d77-9687-8465dc506ca8",
+        "delete_marker": false,
+        "key": "my_file.txt",
+        "size": 14
+    }
+
+Should we want to update the same file with a new version we would use the same command. Automatically, the file will be versioned.
```

Comment:
`If we want...`?

Reply:
https://ell.stackexchange.com/questions/84695/should-versus-if